Do not use *.exapmle files

yuki24 opened the pull request:
I don't think this is a good idea because this is just an extra step to install the app which is not helpful at all. Is there any particular reason to do this?

yuki24 committed:
Do not use *.exapmle files

This is just annoying. An extra step to instal the app that doesn't make
any sense.

m-yamashita commented:
リクエストありがとうございます。
英語を書くのは苦手なので日本語で失礼します。

ご指摘の通り、exampleファイルはエンドユーザーのインストール作業においては無駄なファイルコピーの手間が発生するといった状況です。

このファイルが必要な理由としましては、開発時に間違ってこれらのファイルの設定を残したままコミットされてしまう事態を防ぐという目的があります。

exampleがついたファイルには、データベースのユーザ名及びパスワードや、メールアカウント情報などの個人情報や重要な秘匿情報を扱うものであるため、このように敢えて手順を踏ませ、.gitignoreに追加しておくという方法を採用しています。

他に何か良いソリューションがあれば教えてください。宜しくお願いします！

yuki24 commented:
各個人の credentials や API token がコミットされるのを防ぎたいなら、`ENV` に置き換えてしまうのではダメですかね？

m-yamashita commented:
なるほど、それはいいかもしれません。
採用させていただきたいので、設定済みのファイルをコミットしていただけますか？お手数ですが宜しくお願いします。_(-ω-`_)⌒)_

yuki24 committed:
Do not ingore environment files

yuki24 committed:
Use ENV variables to configure rails app

Also, smtp settings don't have to be configured as long as delivery_method is set to :test, so just remove it.

yuki24 committed:
Add .env and dotenv gem so people can easily manage credentials

yuki24 commented:
更新しました。
`secret_key` の設定は、`.env` を利用することで不要となりますので、README からも削除しました。その他、devise で設定されるべき `secret_key` が空の文字列のままになっていましたので、こちらも設定するように書き直しました。

m-yamashita commented:
素晴らしいです！ありがとうございます！
dotenvというものがあるのですね、勉強になります。

相談なのですが、この場合ですと、
開発環境(local)とproduction環境の設定を分けて管理できないと思うので、そこを解決できますか？

個人的には、環境変数名にDEV_やPROD_等のprefixをつけるくらいしか思いつきませんが…。

yuki24 commented:
環境別に変数を持てば、1台のマシンの上で環境毎に違う設定を持つことができますが、そもそもそのような状況はあるのでしょうか？
多くの場合、開発用のマシンはずっと開発用で、本番用のマシンはずっと本番用の設定で運用されるのではないでしょうか？そのような状況下で環境別の設定を持つメリットは無く、利用者に「なぜ環境毎に変数が用意されているのか？」と不要な思考をさせてしまうだけだと思います。

m-yamashita commented:
確かに、多くの場合を考えるとそうかもしれませんね。

たとえ同じマシン上で動かす場合があったとしても、
別にプロジェクトのディレクトリを切ってそれぞれの.envを置けばよさそうですし、
使って頂く事の方が優先だと考えられますのでこちらの方法で問題ないと思います。
ありがとうございました！

もう一つ、database.example.ymlについては変更されていないようですが、
こちらには何か意図がありますか？

yuki24 committed:
STMP -> SMTP

yuki24 commented:
`config/database.yml` に関しては、特に強い意見はありません。ただ、アダプター毎に設定項目が違うので、少しややこしくなってしまいそうです。`config/database.yml` も `ENV` 使う方針にします？

m-yamashita commented:
そうですね…確かに設定項目が固定ではないので、難しい気がします。

調べてみると、ファイルコピーする事自体はわりとポピュラーな方法ではあるみたいなので、ここは今無理に頑張らなくても良い気がします。ひとまずこれは動作確認の後マージしておこうと思います。

ありがとうございました！

m-yamashita commented:
developブランチへマージしました。 https://github.com/m-yamashita/lodge/commit/338d74b60cf5d5a0c78716a7c16762bbcb57beb2

その後、https://github.com/m-yamashita/lodge/commit/2c3e0cb46d2b714a6266688118d61a35abc06f33 に於いて、MAIL_SENDERを環境変数として設定できるようにしました。

次のバージョン(0.9.2)と共にmasterへマージされますので、こちらは一先ずCloseとさせていただきます。
# こういう時にMerged扱いにするにはどうしたらいいんだろう...。
ありがとうございました！
